util/gpool/spmc/spmcpool.go

```diff
@@ -274,29 +273,35 @@ func (p *Pool[T, U, C, CT, TF]) AddProduceBySlice(producer func() ([]T, error),
 		p.taskManager.AddSubTask(taskID, &taskBox)
 		p.taskCh <- &taskBox
 	}
+	wg.Add(1)
 	go func() {
 		defer func() {
 			if r := recover(); r != nil {
 				logutil.BgLogger().Error("producer panic", zap.Any("recover", r), zap.Stack("stack"))
 			}
-			close(closeCh)
 			close(inputCh)
+			wg.Done()
 		}()
 		for {
-			tasks, err := producer()
-			if err != nil {
-				if errors.Is(err, gpool.ErrProducerClosed) {
+			select {
```

Comment:
this part code seem same with line 342，could we merge them into one func？

Reply:
Done